File: Multiplicacion-clasica(3bucles)/MultiplicacionClasica.cpp
#include <iostream>
#include <vector>
#include <chrono>

using namespace std;
using namespace std::chrono;

// Multiplicacion clasica
void multiplyClassic(int N, vector<vector<double>>& A, vector<vector<double>>& B, vector<vector<double>>& C) {
    auto start = high_resolution_clock::now();

    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            for (int k = 0; k < N; k++) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    auto end = high_resolution_clock::now();
    cout << "Multiplicacion clasica (N=" << N << "): " << duration<double, milli>(end - start).count() << " ms" << endl;
}
int main() {
    // Tamaños de matriz
    vector<int> sizes = {500, 1000, 1500}; 

    for (int N : sizes) {
        vector<vector<double>> A(N, vector<double>(N, 1.0));
        vector<vector<double>> B(N, vector<double>(N, 1.0));
        vector<vector<double>> C(N, vector<double>(N, 0.0));

        multiplyClassic(N, A, B, C);
   }
    return 0;
}

File: Multiplicacion-clasica(3bucles)/Multiplicacionclasica.go
package main

import (
	"fmt"
	"time"
)

func multiplyMatrices(N int) {
	A := make([][]float64, N)
	B := make([][]float64, N)
	C := make([][]float64, N)

	// Inicializar matrices
	for i := range A {
		A[i] = make([]float64, N)
		B[i] = make([]float64, N)
		C[i] = make([]float64, N)
		for j := range A[i] {
			A[i][j] = 1.0
			B[i][j] = 1.0
		}
	}

	start := time.Now()

	// Multiplicacion clasica
	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			for k := 0; k < N; k++ {
				C[i][j] += A[i][k] * B[k][j]
			}
		}
	}

	duration := time.Since(start).Milliseconds()

	fmt.Printf("Tiempo para N = %d: %d ms\n", N, duration)
}

func main() {
  // Tamaños de las matrices
	sizes := []int{500, 1000, 1500} 

	for _, N := range sizes {
		multiplyMatrices(N)
	}
}
